spicy/toolchain/src/compiler/codegen/parsers/literals.cc

```diff
@@ -236,8 +295,61 @@ struct Visitor : public hilti::visitor::PreOrder<Expression, Visitor> {
 } // namespace
 
 Expression ParserBuilder::parseLiteral(const Production& p, const std::optional<Expression>& dst) {
-    if ( auto e = Visitor(this, p, dst).dispatch(p.expression()) )
-        return std::move(*e);
+    switch ( state().literal_mode ) {
+        case LiteralMode::Default:
+        case LiteralMode::Try: {
+            if ( auto e = Visitor(this, p, dst).dispatch(p.expression()) )
+                return std::move(*e);
+
+            hilti::logger().internalError(
+                fmt("codegen: literal parser did not return expression for '%s'", p.expression()));
+
+            hilti::util::cannot_be_reached();
+        }
+
+        case LiteralMode::Search: break;
+    }
+
+    // The following code handles search mode.
+    assert(state().literal_mode == LiteralMode::Search);
+
+    // Set up return value.
+    auto result = builder()->addTmp("result", builder::default_(type::Optional(type::stream::Iterator())));
+
+    // Add a loop for search mode.
+    pushBuilder(builder()->addWhile(builder::bool_(true)), [&]() {
```

Comment:
This loop is in the wrong spot. If we try to match multiple literals in `getLookAhead` we do not want to loop while trying to find each one of them, but loop until either of them is found. That means this loop needs to move into `getLookAhead`.